pkg/querier/replication.go

```diff
@@ -331,11 +338,21 @@ func (r *replicasPerBlockID) blockPlan(ctx context.Context) map[string]*ingestv1
 		smallestCompactionLevel = int32(0)
 	)
 
-	if err := r.pruneIncompleteShardedBlocks(); err != nil {
+	sharded, err := r.pruneIncompleteShardedBlocks()
+	if err != nil {
 		level.Warn(r.logger).Log("msg", "block planning failed to prune incomplete sharded blocks", "err", err)
 		return nil
 	}
-	if err := r.pruneSupersededBlocks(); err != nil {
+
+	// Depending on whether split sharding is used, the compaction level at
+	// which we the data gets deduplicated differs: if split sharding is
+	// enabled, we deduplicate at level 3, and at level 2 otherwise.
```

Comment:
```suggestion
	// Depending on whether split sharding is used, the compaction level at
	// which the data gets deduplicated differs: if split sharding is enabled,
	// we deduplicate at level 3, and at level 2 otherwise.
```